src/Workspaces/Remote/ServiceHub/Services/SourceGeneration/RemoteSourceGenerationService.cs

```diff
@@ -64,11 +71,65 @@ protected override IRemoteSourceGenerationService CreateService(in ServiceConstr
         }, cancellationToken);
     }
 
-    public ValueTask<bool> HasGeneratorsAsync(Checksum solutionChecksum, ProjectId projectId, CancellationToken cancellationToken)
+    private static readonly Dictionary<string, (AnalyzerReferenceMap analyzerReferenceMap, AnalyzerReferenceMap.CreateValueCallback callback)> s_languageToAnalyzerReferenceMap = new()
     {
-        return RunServiceAsync(solutionChecksum, async solution =>
+        { LanguageNames.CSharp, (new(), static analyzerReference => HasSourceGenerators(analyzerReference, LanguageNames.CSharp)) },
+        { LanguageNames.VisualBasic, (new(), static analyzerReference => HasSourceGenerators(analyzerReference, LanguageNames.VisualBasic)) },
```

Comment:
same here.  we can hardcode in C#/vb here to simplify thigns.

Reply:
note: we cache thsi per analyzer reference so that multiple projects with the same analyzer references (common), and lang, will cache the same result.